module3/lessons/mobile_friendly_design_bootstrap.md

```diff
@@ -0,0 +1,208 @@
+
+For example, if we wanted to change the `font-size` property of all `<h1>` elements on a screen that is less than 800 pixels wide: 
+
+```css
+@media only screen and (max-width: 800px) {
+   h1 {
+      font-size: 22px;
+      color: "red";
+      font-family: "Comic Sans MS";
+   }
+}
+```
+
+Using this rule, if our screen is _larger_ than 800px wide, then the `font-size` property stays the same for whatever rules came before it. But as soon as the screen shrinks _below 800px_, the font characteristics of those `<h1>` elements then change to that rule.
+
+<div class="try-it-section">
+<h3>Try it!</h3>
+<p>This page has been outfitted with a similar media query to the one above, only it applies to any `h1`, `h2`, and `h3` tag. </p>
+<p><strong>Shrink and expand your browser window to < 800px wide</strong> to see this query in action.</p>
+</div>
+
+Once you create your application's front-end structure - a section for your navigation, another section for your main body, and a section for your footer - you may need many media queries to account for ALL of the different sizes of devices from which your website may be visited. Instantly, we are overwhelmed by CSS! 🤯
+
+How can we make this easier? 
+
+## Using Bootstrap for Responsive Design
+
+### Columns
+
+A framework like Bootstrap takes these media queries and injects them into **HTML classes** that we can sprinkle around our HTML pages in order to achieve responsiveness. The system that Bootstrap uses is a **rows and columns** approach, where an entire page is **12 columns wide**. We can then divide our page into column widths based on the number of columns it should contain. 
+
+![Columns](./images/bootstrap/columns1.png)
+
+For example, if we have sections for Navigation, Main Content, and a Sidebar, we could say that our column widths would be ***3, 6, and 3***, respectively.
+
+![3 Columns](./images/bootstrap/columns2.png)
+
+We **always** want our columns' widths to add up to **12**, reading them left-to-right. <br>This can be done in any combination you can think of! For this example, we have `3 + 6 + 3` = 12. 
+<br>
+The `3` columns are exactly _half_ the width of the large `6` column, which is also exactly half of the entire 12-column width. 
+
+### Practice: Bootstrap Math
+
+* From left to right, what would the column math look like for 1 small column and 1 large column? 
+  * 3 small columns and 2 medium columns? 
+  * 2 equally-divided columns? 
+* Go to your favorite news website (e.g. cnn.com, nytimes.com, etc.). What might the column widths be for those websites? 
+* Would it be possible to have 3 columns, with the largest column being of `9` width? Why or why not? 
```

Comment:
This is awesome, I wish I'd done this kind of practice math when learning Bootstrap (and css grid concepts, for that matter)